smartdns: new, 35

WhiredPlanck opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

Introduce `smartdns`, a local DNS server to obtain the fastest website IP for better Internet experience.

Package(s) Affected
-------------------

`smartdns`

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

<!-- Yes - Issue Number: ISSUENUMBER -->
<!-- No -->

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

<!--
Build Order
-----------

Please describe in what order maintainers should build this pull request.
-->

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- TODO: CI to auto-fill architectural progress. -->

WhiredPlanck committed:
smartdns: new, 35

WhiredPlanck committed:
smartdns: add conffiles

WhiredPlanck committed:
smartdns: remove unnecessary item from conffiles

WhiredPlanck committed:
smartdns: polish package description

MingcongBai commented on extra-network/smartdns/autobuild/defines:
```diff
@@ -0,0 +1,5 @@
+PKGSEC=net
+PKGDES="A local DNS server to obtain website IP with fastest connection response"
+PKGDEP="openssl"
+BUILDDEP="systemd"
```
Not necessary to include any dependencies already preinstalled with BuildKit.

MingcongBai commented on extra-network/smartdns/autobuild/defines:
```diff
@@ -0,0 +1,5 @@
+PKGNAME=smartdns
+PKGSEC=net
+PKGDES="A local DNS server to obtain website IP with fastest connection response"
```
A local DNS server which resolves domain IP addresses based on the fastest response

WhiredPlanck committed:
smartdns: polish package description

- Credited by MincongBai

WhiredPlanck committed:
smartdns: remove unnecessary BUILDDEP

- Not necessary to include any dependencies already preinstalled with BuildKit.
